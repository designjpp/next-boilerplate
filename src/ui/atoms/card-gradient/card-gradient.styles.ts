import { css } from '@emotion/css';
import xw from 'xwind';

import { tailwindTheme } from 'lib/utils';

const border = {
	blueGradient: `linear-gradient(134.14deg, ${tailwindTheme.theme.colors.blue} 0%, ${tailwindTheme.theme.colors['white-opacity-10']} 25.6%)`,
	blueHoverClassName: xw`hover:bg-blue`,
	yellowGradient: `linear-gradient(146.28deg, ${tailwindTheme.theme.colors.yellow} 0%, ${tailwindTheme.theme.colors['white-opacity-10']} 25.6%)`,
	yellowHoverClassName: xw`hover:bg-yellow`,
};

const background = {
	blue: `linear-gradient(130.76deg, ${tailwindTheme.theme.colors['blue-opacity-10']} 0%, transparent 50.03%), linear-gradient(0deg, ${tailwindTheme.theme.colors['white-opacity-10']}, ${tailwindTheme.theme.colors['white-opacity-10']})`,
	yellow: `linear-gradient(137.94deg, ${tailwindTheme.theme.colors['yellow-opacity-10']} 0%, transparent 30.89%), linear-gradient(0deg, ${tailwindTheme.theme.colors['white-opacity-10']}, ${tailwindTheme.theme.colors['white-opacity-10']})`,
};

const wrapperBase = xw`p-2px rounded-20px transition-colors ease-in-out duration-500`;
export const wrapperClassName = css`
	${wrapperBase}

	&.blue {
		background: ${border.blueGradient};
		${border.blueHoverClassName}
	}
	&.yellow {
		background: ${border.yellowGradient};
		${border.yellowHoverClassName}
	}
`;

const middleBase = xw`bg-dark rounded-20px h-full`;
export const middleClassName = css`
	${middleBase}
`;

const containerBase = xw`bg-white-opacity-10 rounded-20px h-full`;
export const containerClassName = css`
	${containerBase};

	&.blue {
		background: ${background.blue};
	}
	&.yellow {
		background: ${background.yellow};
	}
`;
