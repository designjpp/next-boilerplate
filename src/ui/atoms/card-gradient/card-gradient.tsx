import { cx } from '@emotion/css';

import {
	containerClassName,
	wrapperClassName,
	middleClassName,
} from './card-gradient.styles';

import type { FC } from 'react';

interface CardGradientProps {
	as?: keyof JSX.IntrinsicElements;
	innerAs?: keyof JSX.IntrinsicElements;
	variant: 'blue' | 'yellow';
}

export const CardGradient: FC<CardGradientProps> = ({
	as: OuterAs = 'div',
	innerAs: InnerAs = 'div',
	variant,
	children,
}) => (
	<OuterAs className={cx(wrapperClassName, variant)}>
		<div className={middleClassName}>
			<InnerAs className={containerClassName}>{children}</InnerAs>
		</div>
	</OuterAs>
);
