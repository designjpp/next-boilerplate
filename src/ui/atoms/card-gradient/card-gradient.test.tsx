import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import { create } from 'react-test-renderer';

import { CardGradient } from './card-gradient';

describe('<CardGradient>', () => {
	it("should match is's blue variant snapshot", () => {
		const component = create(<CardGradient variant='blue' />).toJSON();
		expect(component).toMatchSnapshot();
	});

	it("should match is's yellow variant snapshot", () => {
		const component = create(<CardGradient variant='yellow' />).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<CardGradient variant='blue' />);
		const results = await axe(container);
		expect(results).toHaveNoViolations();
	});
});
