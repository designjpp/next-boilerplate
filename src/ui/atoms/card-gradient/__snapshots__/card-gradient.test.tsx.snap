// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardGradient> should match is's blue variant snapshot 1`] = `
.emotion-0 {
  padding: 0.125rem;
  border-radius: 1.25rem;
  transition-property: background-color,border-color,color,fill,stroke;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 500ms;
}

.emotion-0.blue {
  background: linear-gradient(134.14deg, #3AB3DA 0%, rgba(255, 255, 255, 0.1) 25.6%);
}

.emotion-0.blue:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(58, 179, 218, var(--tw-bg-opacity));
}

.emotion-0.yellow {
  background: linear-gradient(146.28deg, #FFC93D 0%, rgba(255, 255, 255, 0.1) 25.6%);
}

.emotion-0.yellow:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(255, 201, 61, var(--tw-bg-opacity));
}

.emotion-1 {
  --tw-bg-opacity: 1;
  background-color: rgba(33, 33, 33, var(--tw-bg-opacity));
  border-radius: 1.25rem;
  height: 100%;
}

.emotion-2 {
  background-color: rgba(255, 255, 255, 0.1);
  border-radius: 1.25rem;
  height: 100%;
}

.emotion-2.blue {
  background: linear-gradient(130.76deg, rgba(58, 179, 218, 0.1) 0%, transparent 50.03%),linear-gradient(0deg, rgba(255, 255, 255, 0.1), rgba(255, 255, 255, 0.1));
}

.emotion-2.yellow {
  background: linear-gradient(137.94deg, rgba(255, 201, 61, 0.1) 0%, transparent 30.89%),linear-gradient(0deg, rgba(255, 255, 255, 0.1), rgba(255, 255, 255, 0.1));
}

<div
  className="emotion-0 blue"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    />
  </div>
</div>
`;

exports[`<CardGradient> should match is's yellow variant snapshot 1`] = `
.emotion-0 {
  padding: 0.125rem;
  border-radius: 1.25rem;
  transition-property: background-color,border-color,color,fill,stroke;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 500ms;
}

.emotion-0.blue {
  background: linear-gradient(134.14deg, #3AB3DA 0%, rgba(255, 255, 255, 0.1) 25.6%);
}

.emotion-0.blue:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(58, 179, 218, var(--tw-bg-opacity));
}

.emotion-0.yellow {
  background: linear-gradient(146.28deg, #FFC93D 0%, rgba(255, 255, 255, 0.1) 25.6%);
}

.emotion-0.yellow:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(255, 201, 61, var(--tw-bg-opacity));
}

.emotion-1 {
  --tw-bg-opacity: 1;
  background-color: rgba(33, 33, 33, var(--tw-bg-opacity));
  border-radius: 1.25rem;
  height: 100%;
}

.emotion-2 {
  background-color: rgba(255, 255, 255, 0.1);
  border-radius: 1.25rem;
  height: 100%;
}

.emotion-2.blue {
  background: linear-gradient(130.76deg, rgba(58, 179, 218, 0.1) 0%, transparent 50.03%),linear-gradient(0deg, rgba(255, 255, 255, 0.1), rgba(255, 255, 255, 0.1));
}

.emotion-2.yellow {
  background: linear-gradient(137.94deg, rgba(255, 201, 61, 0.1) 0%, transparent 30.89%),linear-gradient(0deg, rgba(255, 255, 255, 0.1), rgba(255, 255, 255, 0.1));
}

<div
  className="emotion-0 yellow"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    />
  </div>
</div>
`;
